raft/node.go

```diff
@@ -159,6 +168,9 @@ type Node interface {
 	// commands. For example. when the last Ready contains a snapshot, the application might take
 	// a long time to apply the snapshot data. To continue receiving Ready without blocking raft
 	// progress, it can call Advance before finishing applying the last ready.
+	//
+	// NOTE: Advance must not be called when using AsyncStorageWrites. Response messages from the
```

Comment:
We could do this if we wanted to. The low-level `raft` object actually is entirely async now. It never consults the `asyncStorageWrites` flag. Instead, this flag is only consulted in `node.go` and `rawnode.go` when constructing, accepting, and advancing Ready structs. The question we would need to answer is whether a separate async variant of `RawNode` and `Node` would be worth the code duplication.

Even without a split interface, I think it would be worthwhile to lift this configuration from `raft` to `RawNode` to make its role clearer.

On a related note, I think we should move `newReady` from `node.go` to `rawnode.go` so that async storage writes are entirely handled in `rawnode.go`.

Reply:
> Even without a split interface, I think it would be worthwhile to lift this configuration from raft to RawNode to make its role clearer.

Done.

> On a related note, I think we should move newReady from node.go to rawnode.go so that async storage writes are entirely handled in rawnode.go.

I left a TODO about this that I'll address before merging.

Reply:
> I left a TODO about this that I'll address before merging.

Done.